src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC } from 'react';
+import { Todo } from '../../types/types';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+type Props = {
+  visibleTodos: Todo[];
+  tempTodo: Todo | null;
+  activeIds: number[];
+  handleRemoveTodo: (id: number) => void;
+};
+
+export const TodoList: FC<Props> = ({
+  visibleTodos,
+  tempTodo,
+  activeIds,
+  handleRemoveTodo,
+}) => (
+  <section className="todoapp__main">
+    {visibleTodos.map((todo) => (
+      <TodoItem
+        todo={todo}
+        key={todo.id}
+        activeIds={activeIds}
+        handleRemoveTodo={handleRemoveTodo}
+      />
+    ))}
+    {tempTodo && (
```

Comment:
Consider removing 'tempTodo' and use 'visibleTodos' for both temporary and saved todos. Include the temporary todo in the 'visibleTodos' array and handle its state in the parent component.